### ISSUE TITLE
DT headers added too late for Feign

### ISSUE THREAD

**Issue by jodeev** [newrelic/newrelic-java-agent#249]
[Feign (from Spring Cloud Finchley.SR1)](https://spring.io/projects/spring-cloud-openfeign) appears to be altering the codepath a generated request takes and causes our auto-injection of DT headers to fail.

Description: 
* The application uses Spring Boot (version 2.1.10.RELEASE) and Feign (from Spring Cloud Finchley.SR1).
* When using a rest client, generated by Feign, New Relic agent is adding headers too late: it comes to sun.net.www.protocol.http.HttpURLConnection#addRequestProperty and fails to add any headers with IllegalStateException("Already connected").
* With Spring's RestTemplate the headers are added correctly.

Test app showing this issue included here: 
https://newrelic.zendesk.com/agent/tickets/442571

https://newrelic.aha.io/ideas/ideas/JAVA-I-1001

**Comment by gustavodaquino** [comment]
The problem persists using the following versions:

```
org.springframework.boot:2.6.10
com.newrelic.agent.java:newrelic-api:7.6.0
org.springframework.cloud:spring-cloud-starter-openfeign:3.1.4
io.github.openfeign:feign-httpclient:11.8
```

Regards.

**Comment by kford-newrelic** [comment]
Unfortunately, we do not have the engineering bandwidth to further investigate this issue.

**Comment by jasonjkeller** [comment]
I believe this issue should actually be fixed in Java agent 8.0.1 by https://github.com/newrelic/newrelic-java-agent/pull/1145
